src/backend/parser/parse_relation.c

```diff
@@ -1355,7 +1353,8 @@ addRangeTableEntryForFunction(ParseState *pstate,
 		 * mark this here because this is where we know that the function is being
 		 * used as a RangeTableEntry.
 		 */
-		if (funcexpr && IsA(funcexpr, FuncExpr))
+		Assert(funcexpr);
```

Comment:
Is this patch about pstate?

Reply:
Wrong branch, do you mind if I make force push?

Reply:
yes, you may force-push this branch

Reply:
Done